src/java.desktop/share/classes/javax/swing/JScrollBar.java

```diff
@@ -753,13 +753,46 @@ public void stateChanged(ChangeEvent e)   {
         }
     }
 
-    // PENDING(hmuller) - the next three methods should be removed
+    /**
+     * Unlike most components, JScrollBar will derive the minimum size from
+     * the preferred size in one axis and a fixed minimum size in the other
+     * unless otherwise set by the user.
+     * Thus, it overrides {@code JComponent.setMinimumSize} contract
+     * that subsequent calls to getMinimumSize will return the
+     * same value as set in {@code JComponent.setMinimumSize}
+     *
+     * @param minimumSize the new minimum size of this component
+     */
+    public void setMinimumSize(Dimension minimumSize) {
+        super.setMinimumSize(minimumSize);
+    }
+
+    /**
+     * Unlike most components, JScrollBar will derive the maximum size from
+     * the preferred size in one axis and a fixed maximum size in the other
+     * unless otherwise set by the user.
+     * Thus, it overrides {@code JComponent.setMaximumSize} contract
+     * that subsequent calls to getMaximumSize will return the
+     * same value as set in {@code JComponent.setMaximumSize}
+     *
+     * @param maximumSize the desired maximum allowable size
+     */
+    public void setMaximumSize(Dimension maximumSize) {
+        super.setMaximumSize(maximumSize);
+    }
 
     /**
      * The scrollbar is flexible along it's scrolling axis and
      * rigid along the other axis.
+     *
+     * @return the value of the {@code minimumSize} property if set by user
+     *         or if not set, minimum size derived from
```

Comment:
Seems we are not there yet. What is in this PR now is internally inconsistent.
It adopts doc on set*Size that reads as if get*Size will always ignore the super-class contract,
but on the get*Size methods we have new doc and implementation that actually in large part honours it.
One of these needs to change and then the CSR needs to match.

So, revisiting old ground, ie we've been back and forth on this.
we can keep the current PR code which is "if the user calls set*Size, then honour it, accepting the low risk, OR
we can keep the long standing behaviour, which is as near as we can get to no risk.

I'm still a bit on the fence, but I think I never want to hear about JScrollBar size again, :-)
So can we delete the new code in the get*Size methods and copy some of the wording from set*
     *JScrollBar will derive the maximum size from
      * the preferred size in one axis and a fixed maximum size in the other
     
then the return doc can reduce to
@ return the minimum size calculated as described above.

Reply:
> So can we delete the new code in the get_Size methods and copy some of the wording from set_ *JScrollBar will derive the maximum size from * the preferred size in one axis and a fixed maximum size in the other
> 
> then the return doc can reduce to @ return the minimum size calculated as described above.

Updated as per suggestion..